File: user/src/main/java/io/philo/support/PasswordEncoder.java
package io.philo.support;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public abstract class PasswordEncoder {

  private static final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();

  public static String encodePassword(String rawPassword) {
    return encoder.encode(rawPassword);
  }

  public static boolean isSamePassword(String rawPassword, String encodedPassword) {
    return encoder.matches(rawPassword, encodedPassword);
  }
}

File: common/security/src/main/kotlin/io/philo/shop/PasswordEncoder.kt
package io.philo.shop

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

/**
 * Facade Of Password Encoding
 */
object PasswordEncoder {
    private val encoder = BCryptPasswordEncoder()
    fun encodePassword(rawPassword: String?): String {
        return encoder.encode(rawPassword)
    }

    fun isSamePassword(rawPassword: String?, encodedPassword: String?): Boolean {
        return encoder.matches(rawPassword, encodedPassword)
    }
}
